src/picongpu/include/particles/ionization/byField/BSI/AlgorithmBSI.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2014 Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "types.h"
+#include "particles/ionization/ionizationEnergies.param"
+
+/* IONIZATION ALGORITHM */
+
+namespace picongpu
+{
+    namespace particles
+    {
+        namespace ionization
+        {
+            
+            struct AlgorithmBSI
+            {
+
+                template<typename EType, typename BType, typename ParticleType >
+                        __host__ DINLINE void operator()(
+                                                            const BType bField, /* at t=0 */
+                                                            const EType eField, /* at t=0 */
+                                                            ParticleType& parentIon) /* with position at t=0 and momentum at t=-1/2 */
+                {
+
+                    /*Barrier Suppression Ionization for hydrogenlike helium */
+                    if (math::abs(eField)*UNIT_EFIELD >= SI::IONIZATION_EFIELD && parentIon[chargeState_] < 2)
```

Comment:
Okay.

For really implementing the model I will need some sort of storage matrix or even matrix-like object. It should connect the yet-to-come attribute `protonNumber` to the maximum number of entries in a column/line and for each charge state there should be a corresponding binding energy that will be internally converted to an electric field. This will be the field needed for ionizing to the next charge state.